### PR TITLE
Add compareValues prop to useSelect's reducer

### LINKED ISSUE #1401: Options with "value" field as object or array type are duplicated
**Scope**: PF4 mapper, Form renderer

**Description**

How to reproduce: use a schema with `component` set to "select", and `isSearchable` to "true" with `loadOptions` function (so that you are able to asynchronously load values for the select component. Also, in `loadOptions` return a promise that resolves in an array with options. The options have to have value of array or object type, not primitive (like string or numbers).

Current behavior: the select component duplicates the value as you type in the select component and new data loads.

[Screencast from 2023-08-22 18-01-52.webm](https://github.com/data-driven-forms/react-forms/assets/31385370/b656cb15-7545-4aff-8903-a133e4abb60d)

Expected behavior: the select component renders just only the options it gets from `loadOptions`.

**Schema**

```
const loadOptions = (searchValue = '') => {
  return getGroups({ name: searchValue }).then((data) => {
    return data.results.map(({ name, id }) => ({
      label: name,
      value: { id, name },
    }));
  });
};

const exampleSchema = (hosts) => ({
  fields: [
    {
      component: 'select',
      name: 'group',
      label: 'Select a group',
      simpleValue: true,
      isSearchable: true, // enables typeahead
      isRequired: true,
      isClearable: true,
      placeholder: 'Type or click to select a group',
      loadOptions,
      options: [],
      validate: [{ type: validatorTypes.REQUIRED }],
    }
  ],
});
```

### PR DESCRIPTION
**Pull request by victoria-dos**
Fixes #1401

**Description**

**Schema** *(if applicable)*

```jsx
const loadOptions = (searchValue = '') => {
  return getGroups({ name: searchValue }).then((data) => {
    return data.results.map(({ name, id }) => ({
      label: name,
      value: { id, name },
    }));
  });
};

const exampleSchema = (hosts) => ({
  fields: [
    {
      component: 'select',
      name: 'group',
      label: 'Select a group',
      simpleValue: true,
      isSearchable: true, // enables typeahead
      isRequired: true,
      isClearable: true,
      placeholder: 'Type or click to select a group',
      loadOptions,
      options: [],
      validate: [{ type: validatorTypes.REQUIRED }],
    }
  ],
});
```

**Checklist:** *(please see [documentation page](https://data-driven-forms.org/development-setup) for more information)*

- [x] `Yarn build` passes
- [x] `Yarn lint` passes
- [x] `Yarn test` passes
- [x] Test coverage for new code *(if applicable)*
- [x] Documentation update *(if applicable)*
- [x] Correct commit message
   - format `fix|feat({scope}): {description}`
   - i.e. `fix(pf3): wizard correctly handles next button`
   - fix will release a new \_.\_.X version
   - feat will release a new \_.X.\_ version (use when you introduce new features)
     - we want to avoid any breaking changes, please contact us, if there is no way how to avoid them
   - scope: package
   - if you update the documentation or tests, do not use this format
     - i.e. `Fix button on documenation example page`

### CODE REVIEW

File: packages/common/src/tests/select/select.test.js [hunk 1]
```diff
@@ -501,7 +502,149 @@ describe('Select test', () => {
...
+                },
+              ],
+            }}
+          />
+        );
+      });
+
+      // screen.debug(screen.getByRole('form'));
+
+      // expect(state.value).toEqual([{ label: 'first' }]);
```
Comment (rvsia) [inline packages/common/src/tests/select/select.test.js:537]: comment leftovers

File: packages/common/src/select/select.js [hunk 2]
```diff
@@ -5,6 +5,36 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import useSelect from '../use-select/use-select';
 
+const deepEqual = (a, b, depth = 0) => {
```
Comment (rvsia) [inline packages/common/src/select/select.js:8]: Could we provide some unit tests for this function and extract it to a separate file?

Comment (Hyperkid123) [reply packages/common/src/select/select.js:8]: Sure

**Review by rvsia — APPROVED** [review]

**Pull request merged**